src/rpc/blockchain.cpp

```diff
@@ -2749,6 +2749,11 @@ static RPCHelpMan loadtxoutset()
     }
 
     SnapshotMetadata metadata;
+    // Check if the file size is at least as large as the expected size of SnapshotMetadata
+    if (afile.size() < sizeof(SnapshotMetadata)) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("Unable to load UTXO snapshot, "
```

Comment:
nit: should we specify that the file size is too small? A user might think their file is too large with this error message

Reply:
just saw your alternative, that is preferred for me

Reply:
My intention with the code change alternative was to avoid touching both stream files, header and .cpp, but sure, I could take the error message from there and use it here.
Regarding the error message output, I'm open to suggestions there, I could specify that the file's too small or the exact minimum amount of bytes that are required, or even ask to check the documentation, which will need to be updated.